### PR TITLE
feat: CMake port media/trick_play

### PR DESCRIPTION
**Pull request by zdanek**
Issue #1047 (CMake porting)

### CODE REVIEW

File: packager/media/base/media_handler_test_base.h [hunk 1]
```diff
@@ -145,8 +146,10 @@ MATCHER_P6(MatchEncryptionConfig,
            constant_iv,
            key_id,
            "") {
-  const std::string constant_iv_hex = absl::BytesToHexString(arg.constant_iv);
-  const std::string key_id_hex = absl::BytesToHexString(arg.key_id);
+  const std::string constant_iv_data(constant_iv.data(), constant_iv.size());
```
Comment (joeyparrish) [inline packager/media/base/media_handler_test_base.h:149]: #1143 has a new utility to make this work without copying the data first.  Let's land that PR, then rework this section to use that utility.

See https://github.com/shaka-project/shaka-packager/pull/1143/files#diff-14bc2abce24d1762d3fbf330540b87e2ae6b573103e47aeffbf2a87e89dde806

File: packager/media/trick_play/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Copyright 2022 Google LLC. All rights reserved.
+#
+# Use of this source code is governed by a BSD-style
+# license that can be found in the LICENSE file or at
+# https://developers.google.com/open-source/licenses/bsd
+
+add_library(media_trick_play STATIC
```
Comment (joeyparrish) [inline packager/media/trick_play/CMakeLists.txt:7]: Please delete the corresponding .gyp file as well.  Thanks!

**Review by joeyparrish — CHANGES_REQUESTED** [review]

File: packager/media/base/media_handler_test_base.h (hunk 1 above)
Comment (zdanek) [reply packager/media/base/media_handler_test_base.h:149]: Sure

File: packager/media/trick_play/CMakeLists.txt (hunk 2 above)
Comment (zdanek) [reply packager/media/trick_play/CMakeLists.txt:7]: Yeah, I forgot 🙂

**Comment by joeyparrish** [comment]
I made several updates to the PR, and merged in other changes from the cmake branch.  The build fails on Windows due to linker errors, but I'm not sure why yet.

**Comment by joeyparrish** [comment]
Working on this again.  Debugging the Windows linker errors from the Shaka lab's Windows machine.  I suspect a misconfiguration of protobuf somehow.  It appears that absl is linked with a different MSVC runtime than what protoc's main.cc wants to use, which leads to the linker error.

**Pull request merged**